mne/inverse_sparse/mxne_optim.py

```diff
@@ -276,20 +277,24 @@ def _mixed_norm_solver_prox(M, G, alpha, lipschitz_constant, maxit=200,
         else:
             R = GTM - np.dot(gram[:, Y_as], Y[Y_as])
 
-        _, p_obj, d_obj, _ = dgap_l21(M, G, X, active_set, alpha, n_orient)
-        highest_d_obj = max(d_obj, highest_d_obj)
-        gap = p_obj - highest_d_obj
-        E.append(p_obj)
-        logger.debug("p_obj : %s -- gap : %s" % (p_obj, gap))
-        if gap < tol:
-            logger.debug('Convergence reached ! (gap: %s < %s)' % (gap, tol))
-            break
+        if (i + 1) % dgap_freq == 0:
+            _, p_obj, d_obj, _ = dgap_l21(M, G, X, active_set, alpha,
+                                           n_orient)
+            highest_d_obj = max(d_obj, highest_d_obj)
+            gap = p_obj - highest_d_obj
+            E.append(p_obj)
+            logger.debug("p_obj : %s -- gap : %s" % (p_obj, gap))
+            if gap < tol:
+                logger.debug('Convergence reached ! (gap: %s < %s)' % (gap,
+                             tol))
+                break
     return X, active_set, E
 
 
 @verbose
 def _mixed_norm_solver_cd(M, G, alpha, lipschitz_constant, maxit=10000,
-                          tol=1e-8, verbose=None, init=None, n_orient=1):
+                          tol=1e-8, verbose=None, init=None, n_orient=1,
+                          dgap_freq=10):
```

Comment:
We should inform the user that dgap_freq is igrnored, when using sklearn's cd solver